pkg/controller/pravegacluster/upgrade.go

```diff
@@ -315,6 +315,9 @@ func (r *ReconcilePravegaCluster) checkUpdatedPods(sts *appsv1.StatefulSet, vers
 
 		if !util.IsPodReady(pod) {
 			// At least one updated pod is still not ready
+			if pod.Status.ContainerStatuses[0].State.Waiting.Reason == "ImagePullBackOff" {
+				return false, fmt.Errorf("pod %s update failed because of ImagePullBackOff", pod.Name)
```

Comment:
It would be good to print the reason using `fmt.Errorf("pod %s update failed because of %s", pod.Name, pod.Status.ContainerStatuses[0].State.Waiting.Reason)`

Reply:
Included this change

Reply:
Ok